lib/exercises.rb

```diff
@@ -1,19 +1,44 @@
 
 # This method will return an array of arrays.
 # Each subarray will have strings which are anagrams of each other
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n^2)
+# Space Complexity: o(n)
 
 def grouped_anagrams(strings)
-  raise NotImplementedError, "Method hasn't been implemented yet!"
+  # raise NotImplementedError, "Method hasn't been implemented yet!"
+  return [] if strings.empty?
+    hash = Hash.new([])
+    strings.each.with_index do |element, i| 
+      hash[element.downcase.chars.sort.join] += [i] 
+    end
+    # p hash
+    results = hash.map do |key, values| 
+      values.map do |i| 
+        strings[i]
+      end
+    end
+    return results
 end
 
 # This method will return the k most common elements
 # in the case of a tie it will select the first occuring element.
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n)
+# Space Complexity: O(n)
 def top_k_frequent_elements(list, k)
```

Comment:
👍 
 
Nice use of sort, I would say that the time complexity is O(n log n) because you're sorting `hash.values`.